packages/widget/src/controllers/transfers/fungible-token-transfer.ts

```diff
@@ -504,73 +504,107 @@ export class FungibleTokenTransferController implements ReactiveController {
     ) {
       return;
     }
+
     switch (this.sourceNetwork.type) {
       case Network.EVM:
         {
-          void this.buildEvmTransactions();
+          //we already check that but this prevents those typescript errors
+          const address = this.walletContext.value?.evmWallet?.address;
+          const provider = this.walletContext.value?.evmWallet?.provider;
+          const providerChainId =
+            this.walletContext.value?.evmWallet?.providerChainId;
+          if (
+            !this.sourceNetwork?.chainId ||
+            !this.walletContext.value?.evmWallet?.provider ||
+            !this.walletContext.value.evmWallet.address ||
+            !address ||
+            !provider ||
+            providerChainId !== this.sourceNetwork.chainId
+          ) {
+            this.estimatedGas = undefined;
+            this.resetFee();
+            return;
+          }
+          const {
+            pendingEvmApprovalTransactions,
+            pendingTransferTransaction,
+            fee,
+            resourceAmount
+          } = await this.buildEvmTransactions({
+            address,
+            chainId: this.destinationNetwork.chainId,
+            destinationAddress: this.destinationAddress,
+            resourceId: this.selectedResource.resourceId,
+            resourceAmount: this.resourceAmount,
+            provider,
+            providerChainId,
+            env: this.env,
+            fee: this.fee,
+            pendingEvmApprovalTransactions: this.pendingEvmApprovalTransactions,
+            pendingTransferTransaction: this
+              .pendingTransferTransaction as UnsignedTransaction,
+            sourceNetwork: this.sourceNetwork
+          });
+
+          this.fee = fee;
+          this.resourceAmount = resourceAmount;
+
+          this.pendingEvmApprovalTransactions = pendingEvmApprovalTransactions;
+          this.pendingTransferTransaction = pendingTransferTransaction;
+
+          this.estimatedGas = await estimateEvmGas(
+            this.getTransferState(),
+            this.sourceNetwork?.chainId,
+            this.walletContext.value?.evmWallet?.provider,
+            this.walletContext.value.evmWallet.address,
+            this.pendingEvmApprovalTransactions,
```

Comment:
`pendingEvmApprovalTransactions` and `pendingTransferTransaction` can be combined into one and this method doesn't need transfer state param

Reply:
solved as per conversation in the office